src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -404,35 +404,22 @@ export class SavedObjectsRepository {
         }
 
         const { requestedId, rawMigratedDoc, esRequestIndex } = expectedResult.value;
-        const response = bulkResponse.items[esRequestIndex];
-        const {
-          error,
-          _id: responseId,
-          _seq_no: seqNo,
-          _primary_term: primaryTerm,
-        } = Object.values(response)[0] as any;
-
-        const {
-          _source: { type, [type]: attributes, references = [], namespaces },
-        } = rawMigratedDoc;
-
-        const id = requestedId || responseId;
+        const { error, ...rawResponse } = Object.values(
+          bulkResponse.items[esRequestIndex]
+        )[0] as any;
+
         if (error) {
           return {
-            id,
-            type,
-            error: getBulkOperationError(error, type, id),
+            id: requestedId,
+            type: rawMigratedDoc._source.type,
+            error: getBulkOperationError(error, rawMigratedDoc._source.type, requestedId),
           };
         }
-        return {
-          id,
-          type,
-          ...(namespaces && { namespaces }),
-          updated_at: time,
-          version: encodeVersion(seqNo, primaryTerm),
-          attributes,
-          references,
-        };
+
+        // When method == 'index' the bulkResponse doesn't include the indexed
+        // _source so we return rawMigratedDoc but have to spread rawResponse
+        // to return the latest _seq_no and _primary_term values from ES.
+        return this._serializer.rawToSavedObject({ ...rawMigratedDoc, ...rawResponse });
```

Comment:
Do we only need `_seq_no` and `_primary_term` from the response? 

If so, maybe this should be explicit instead of spreading the whole `rawResponse` ?
But it seems it's not, as the `migrationVersion` attribute seems to now be present due to this change, so maybe the comment should be changed?

Reply:
migrationVersion is present in the rawMigratedDoc already but we didn't previously return the whole object but instead constructed a custom "serialized" object which didn't include the migrationVersion. We also need the id field since `requestedId` could be undefined (when bulkCreate is called with objects without id's). Generating a new ID is currently hidden in the serialization layer and I think it will be better to make this explicit. So I'll see if I can simplify this code and make it clearer which properties we need from rawResponse instead of spreading the whole object.